src/qudi/hardware/ni_x_series/ni_x_series_finite_sampling_input.py

```diff
@@ -424,7 +425,9 @@ def get_buffered_samples(self, number_of_samples=None):
                     timeout=self._rw_timeout)
                 if read_samples != number_of_samples:
                     return data
-                data[reader._task.name.split('_')[-1]] = data_buffer
+                data[reader._task.name.split('_')[-1]] = data_buffer * self._sample_rate
```

Comment:
To avoid another copy of the `data_buffer` array, it would be better to split this line into two statements, a multiplication (in-place array operation) and the assignment to the `data` dict, e.g.:
```python
data_buffer *= self._sample_rate
data[reader._task.name.split('_')[-1]] = data_buffer
```

The buffer handling is already pretty bad/taxing (written by myself). The NI in-streamer I recently wrote performs much better.

Reply:
Thanks for this feedback, is implemented that way in https://github.com/Ulm-IQO/qudi-iqo-modules/commit/f1eb30f51840e572aadeb8bb49df7abbf00add44.